metagpt/actions/intent_detect.py

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+This script is designed to classify intentions from complete conversation content.
+
+Usage:
+    This script can be used to classify intentions from a conversation. It utilizes models for detecting intentions
+    from the text provided and categorizes them accordingly. If the intention of certain words or phrases is unclear,
+    it prompts the user for clarification.
+
+Dependencies:
+    This script depends on the metagpt library, pydantic, and other utilities for message parsing and interaction.
+
+"""
+import json
+from typing import List
+
+from pydantic import BaseModel, Field
+
+from metagpt.actions import Action
+from metagpt.logs import logger
+from metagpt.schema import Message
+from metagpt.utils.common import parse_json_code_block
+
+
+class SOPItem(BaseModel):
+    """
+    Represents an item in a Standard Operating Procedure (SOP).
+
+    Attributes:
+        description (str): The description or title of the SOP.
+        sop (List[str]): The steps or details of the SOP.
+    """
+
+    description: str
+    sop: List[str]
+
+
+SOP_CONFIG = [
+    SOPItem(
+        description="Intentions related to or including software development, such as developing or building software, games, programming, app, websites, etc.",
+        sop=[
+            "Writes a PRD based on software requirements.",
+            "Writes a design to the project repository, based on the PRD of the project.",
+            "Writes a project plan to the project repository, based on the design of the project.",
+            "Writes codes to the project repository, based on the project plan of the project.",
+            "Run QA test on the project repository.",
+            "Stage and commit changes for the project repository using Git.",
+        ],
+    )
+]
+
+
+class _IntentDetectIntention(BaseModel):
+    """
+    Represents detected intentions.
+
+    Attributes:
+        ref (str): The reference to the original words.
+        intent (str): The detected intention of the referenced words.
+    """
+
+    ref: str
+    intent: str
+
+
+class IntentDetectClarification(BaseModel):
+    """
+    Represents clarifications for unclear intentions.
+
+    Attributes:
+        ref (str): The reference to the original words.
+        clarification (str): A question for the user to clarify the intention of the unclear words.
+    """
+
+    ref: str
+    clarification: str
+
+
+class _IntentDetectDialogIntentions(BaseModel):
```

Comment:
maybe "IntentDetectDialogIntentions" is better

Reply:
done. It has been moved into `IntentDetect`